crates/ironrdp-cliprdr/src/pdu/format_list.rs

```diff
@@ -6,11 +6,164 @@ use ironrdp_pdu::{cast_int, ensure_size, invalid_message_err, PduDecode, PduEnco
+    /// information about the files by passing the handle to the DragQueryFile function.
+    pub const CF_HDROP: Self = Self(15);
+
+    /// The data is a handle (HGLOBAL) to the locale identifier (LCID) associated with text in the
+    /// clipboard. When you close the clipboard, if it contains CF_TEXT data but no CF_LOCALE data,
+    /// the system automatically sets the CF_LOCALE format to the current input language. You can
+    /// use the CF_LOCALE format to associate a different locale with the clipboard text. An
+    /// application that pastes text from the clipboard can retrieve this format to determine which
+    /// character set was used to generate the text. Note that the clipboard does not support plain
+    /// text in multiple character sets. To achieve this, use a formatted text data type such as
+    /// RTF instead.The system uses the code page associated with CF_LOCALE to implicitly convert
+    /// from CF_TEXT to CF_UNICODETEXT. Therefore, the correct code page table is used for the
+    /// conversion.
+    pub const CF_LOCALE: Self = Self(16);
+
+    /// A memory object containing a BITMAPV5HEADER structure followed by the bitmap color space
+    /// information and the bitmap bits.
+    pub const CF_DIBV5: Self = Self(17);
+
+    /// Creates new `ClipboardFormatId` with given id. Note that [`ClipboardFormatId`] already
+    /// defines constants for standard clipboard formats, [`Self::new`] should only be
+    /// used for custom/OS-specific formats.
+    pub fn new(id: u32) -> Self {
+        Self(id)
+    }
+
+    pub fn value(&self) -> u32 {
+        self.0
+    }
+}
+
+/// Clipboard format name. Hardcoded format names defined by [MS-RDPECLIP] are available as
+/// constants.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ClipboardFormatName(Cow<'static, str>);
+
+impl ClipboardFormatName {
+    /// Special format name for file lists defined by [`MS-RDPECLIP`] which is used for clipboard
+    /// data  with [`crate::pdu::format_data::PackedFileList`] payload.
+    pub const FILE_LIST: Self = Self::new_static("FileGroupDescriptorW");
+
+    pub fn new(name: impl Into<Cow<'static, str>>) -> Self {
+        Self(name.into())
+    }
+
+    /// Same as [`Self::new`], but for `'static` string - it can be used in const contexts.
+    pub const fn new_static(name: &'static str) -> Self {
+        Self(Cow::Borrowed(name))
+    }
+
+    pub fn value(&self) -> &str {
+        &self.0
+    }
+}
+
 /// Represents `CLIPRDR_SHORT_FORMAT_NAME` and `CLIPRDR_LONG_FORMAT_NAME`
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct ClipboardFormat {
-    pub id: u32,
-    pub name: String,
+    id: ClipboardFormatId,
+    name: Option<ClipboardFormatName>,
+}
+
+impl ClipboardFormat {
+    /// Creates unnamed `ClipboardFormat` with given id.
+    pub const fn new(id: ClipboardFormatId) -> Self {
+        Self { id, name: None }
+    }
+
+    /// Sets clipboard format name. If specified name is empty, then name will be set to `None`.
```

Comment:
I imagined something more like
```suggestion
    /// Sets clipboard format name.
    ///
    /// This is typically used for custom/OS-specific formats where a name must be associated to
    /// the `ClipboardFomatId` in order to distinguish between vendors.
```

No need to mention how it’s handled internally. Feel free to change if this does not reflect the actual purpose of format names.

Reply:
done